### PR TITLE
Switch default JS engine to Hermes

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-7163

# How

Update `@expo/config-plugins`, `@expo/dev-server`, all templates and `packages/expo-manifest` to use Hermes as the default JS engine 

Follow ups
- [x] Update `Using Hermes Engine` guide (https://github.com/expo/expo/pull/21047)
- [x] Update xdl schema (https://github.com/expo/universe/pull/11355)

# Test Plan

```bash
# create a new project
npx create-expo-app hermes-app
cd hermes-app
# run local expo cli
nexpo prebuild --no-install
```

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/bare-expo/android/app/build.gradle [hunk 1]
```diff
@@ -61,7 +61,7 @@ react {
 
 // Override `hermesEnabled` by `expo.jsEngine`
 ext {
-  hermesEnabled = (findProperty('expo.jsEngine') ?: "jsc") == "hermes"
+  hermesEnabled = (findProperty('expo.jsEngine') ?: "hermes") == "hermes"
```
Comment (gabrieldonadel) [inline apps/bare-expo/android/app/build.gradle:64]: Already enabled on iOS https://github.com/expo/expo/blob/abb8336f09b2d795ba2f91f8f54e441527457f2a/apps/bare-expo/ios/Podfile.properties.json#L2

**Comment by gabrieldonadel** [comment]
Holding on to this until we are sure about making Hermes the default.

File: packages/@expo/dev-server/src/HermesBundler.ts [hunk 2]
```diff
@@ -18,13 +18,23 @@ export function isEnableHermesManaged(expoConfig: ExpoConfig, platform: Platform
...
+
       return (expoConfig.android?.jsEngine ?? expoConfig.jsEngine) === 'hermes';
     }
     case 'ios': {
       if (!gteSdkVersion(expoConfig, '43.0.0')) {
         // Hermes on iOS is supported after SDK 43.
         return false;
       }
+      if (gteSdkVersion(expoConfig, '48.0.0')) {
+        // Hermes on Android is enabled by default after SDK 48.
```
Comment (reichhartd) [inline packages/@expo/dev-server/src/HermesBundler.ts:34]:
```suggestion
        // Hermes on iOS is enabled by default after SDK 48.
```

**Comment by Kudo** [comment]
maybe also update the Podfile
https://github.com/expo/expo/blob/4db800d70bd878ca2f3ec0797ad2e38bb41a4cc6/templates/expo-template-bare-minimum/ios/Podfile#L30

```ruby
:hermes_enabled => podfile_properties['expo.jsEngine'] == nil || podfile_properties['expo.jsEngine'] == 'hermes', 
```

**Comment by gabrieldonadel** [comment]
> maybe also update the Podfile
> 
> https://github.com/expo/expo/blob/4db800d70bd878ca2f3ec0797ad2e38bb41a4cc6/templates/expo-template-bare-minimum/ios/Podfile#L30
> 
> ```ruby
> :hermes_enabled => podfile_properties['expo.jsEngine'] == nil || podfile_properties['expo.jsEngine'] == 'hermes', 
> ```

Thanks for the review, I've just pushed a commit updating all Podfiles to use this logic

**Comment by douglowder** [comment]
@gabrieldonadel The fixes for updates E2E are now working and allow the test app to build and run the tests. A few tests are failing, so I reverted to JSC and will investigate the test failures separately (ENG-7449).

**Comment by Kudo** [comment]
@douglowder what hermes problem did you encounter? does it relate to sourcemap generation? https://github.com/expo/expo/pull/21022

**Comment by douglowder** [comment]
@Kudo I did hit that issue. There were a few others specific to the updates E2E.

**Comment by Kudo** [comment]
@douglowder would like to know about other problems you encountered when you get a chance, that would make sure we are good to ship hermes by default in sdk 48. so it's important for us to know whether there are any incompatible issues from expo-updates.

**Review by Kudo — APPROVED** [review]
i'm approving this pr. once #21045 landed and we confirm hermes performance regression is resolved. we are good to go for this pr 🚀

**Review by tsapeta — APPROVED** [review]

**Pull request merged**